src/Build/Utilities/EngineFileUtilities.cs

```diff
@@ -152,8 +219,100 @@ FileMatcher fileMatcher
                 // the list into a string array.  If the filespec started out
                 // as a relative path, we will get back a bunch of relative paths.
                 // If the filespec started out as an absolute path, we will get
-                // back a bunch of absolute paths.
-                fileList = fileMatcher.GetFiles(directoryUnescaped, filespecUnescaped, excludeSpecsUnescaped);
+                // back a bunch of absolute paths. Also retrieves the search action
+                // and relevant Exclude filespec for drive enumerating wildcard detection.
+                (fileList, action, excludeFileSpec) = fileMatcher.GetFiles(directoryUnescaped, filespecUnescaped, excludeSpecsUnescaped);
+
+                // Determines whether Exclude filespec or passed in file spec should be
+                // used in drive enumeration warning or exception.
+                bool excludeFileSpecIsUnusable = string.IsNullOrWhiteSpace(excludeFileSpec);
```

Comment:
excludeFileSpecIsEmpty? I had no idea what you meant by Unusable.

Reply:
Changing to `excludeFileSpecIsEmpty`, the comments clarify the meaning behind "Unusable," but Empty is better.